vm/src/stdlib/io.rs

```diff
@@ -115,40 +128,50 @@ impl BufferedIO {
         self.cursor.position()
     }
 
-    fn readline(&mut self, size: OptionalOption<i64>, vm: &VirtualMachine) -> PyResult<Vec<u8>> {
-        let mut buf = Vec::new();
-        let mut left = size
-            .flatten()
-            .filter(|s| *s >= 0)
-            .and_then(|s| s.try_into().ok())
-            .unwrap_or(usize::MAX);
-        if left == 0 {
-            return Ok(buf);
-        }
-        loop {
-            let (done, used) = {
-                let mut available = self.cursor.fill_buf().map_err(|err| os_err(vm, err))?;
-                if left < available.len() {
-                    available = &available[..left];
-                }
+    fn readline(&mut self, size: Option<usize>, vm: &VirtualMachine) -> PyResult<Vec<u8>> {
+        let size = match size {
+            None => {
+                let mut buf = String::new();
+                self.cursor
+                    .read_line(&mut buf)
+                    .map_err(|err| os_err(vm, err))?;
+                return Ok(buf.into_bytes());
+            }
+            Some(0) => {
+                return Ok(Vec::new());
+            }
+            Some(size) => size,
+        };
 
-                match available.find_byte(b'\n') {
-                    Some(i) => {
-                        buf.extend_from_slice(&available[..=i]);
-                        (true, i + 1)
-                    }
-                    _ => {
-                        buf.extend_from_slice(available);
-                        (false, available.len())
-                    }
-                }
-            };
-            self.cursor.consume(used);
-            left -= used;
-            if done || used == 0 || left == 0 {
-                return Ok(buf);
+        let available = {
+            let buf = self.cursor.fill_buf().map_err(|err| os_err(vm, err))?;
+            if size < buf.len() {
+                &buf[..size]
+            } else {
+                buf
             }
-        }
+        };
+        let buf = match available.find_byte(b'\n') {
+            Some(i) => (available[..=i].to_vec()),
+            _ => (available.to_vec()),
+        };
+        self.cursor.consume(buf.len());
+        Ok(buf)
+    }
+
+    fn truncate(&mut self, pos: Option<isize>, vm: &VirtualMachine) -> PyResult<()> {
```

Comment:
For consistency with other methods it's probably better to accept `Option<usize>` here and move the check for negative values into the caller, which is what is responsible to convert values from Python into this underlying interface.

Reply:
Agree. I want to add a new method to `OptionalSize`, but I have no idea about the method name.
Do you have any suggestion not to be confused with `to_usize`?

Reply:
I'm also not sure amount the name, maybe `try_to_positive_usize`?

Reply:
Well, usize is always positive anyway 🙃. But yeah, maybe just `to_size()` or something? It is a `[Optional]Size` anyway

Reply:
I used `try_usize`